bin/dhttpd.dart

```diff
@@ -22,8 +22,21 @@ Future<void> main(List<String> args) async {
   await Dhttpd.start(
     path: options.path,
     port: options.port,
+    headers:
+        options.headers != null ? parseKeyValuePairs(options.headers!) : null,
     address: options.host,
   );
 
   print('Server started on port ${options.port}');
 }
+
+Map<String, String> parseKeyValuePairs(String str) {
+  final regex = RegExp(r'([\w-]+)=([\w-]+)(;|$)');
+  final map = <String, String>{};
+  for (final match in regex.allMatches(str)) {
```

Comment:
Could be one line with `Map.fromIterable` or similar

Reply:
I'm trying to make this work like this:

```
  return Map<String, String>.fromIterable(regex.allMatches(str),
      key: (match) => match.group(1) as String,
      value: (match) => match.group(2) as String);
```

It's a little strange, because fromIterable doesn't take a generic type annotation so it can't infer that the type is `Match` in the `key` and `value` callbacks.

This also triggers this lint: https://dart.dev/tools/linter-rules/prefer_for_elements_to_map_fromIterable

I think I'd prefer to keep this as a for loop as per the lint.

Reply:
Actually, the lint is saying to use a collection-for - even better!